### PR TITLE
Update for .NET 5

### LINKED ISSUE #22846: Problem in the Code
[Whats up i tried making the code work in my pc and the file path when i run this line  
   dynamic rFile = new ReadOnlyFile(@"..\..\TextFile1.txt");
from the example make a problem when i go over the walkthrough
but when i changed it to have one more "..\ " it worked fine

    dynamic rFile = new ReadOnlyFile(@"..\..\..\TextFile1.txt");

 ]

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: f63fb701-91e9-5723-add6-33d2288431a3
* Version Independent ID: e9d33444-b5f7-96d4-9c33-dd29913349d5
* Content: [Walkthrough: Creating and Using Dynamic Objects (C# and Visual Basic)](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/types/walkthrough-creating-and-using-dynamic-objects)
* Content Source: [docs/csharp/programming-guide/types/walkthrough-creating-and-using-dynamic-objects.md](https://github.com/dotnet/docs/blob/master/docs/csharp/programming-guide/types/walkthrough-creating-and-using-dynamic-objects.md)
* Product: **dotnet-csharp**
* Technology: **csharp-fundamentals**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### PR DESCRIPTION
**Pull request by tdykstra**
Fixes #22846

### CODE REVIEW

File: docs/csharp/programming-guide/types/walkthrough-creating-and-using-dynamic-objects.md [hunk 1]
```diff
@@ -1,15 +1,14 @@
...
-ms.date: 07/20/2015
+ms.date: 03/24/2021
 dev_langs:
   - "csharp"
   - "vb"
 helpviewer_keywords:
   - "dynamic objects [Visual Basic]"
   - "dynamic objects"
   - "dynamic objects [C#]"
-ms.assetid: 568f1645-1305-4906-8625-5d77af81e04f
```
Comment (gewarren) [inline docs/csharp/programming-guide/types/walkthrough-creating-and-using-dynamic-objects.md:12]: 👋

File: docs/csharp/programming-guide/types/walkthrough-creating-and-using-dynamic-objects.md [hunk 2]
```diff
@@ -21,76 +20,88 @@ Dynamic objects expose members such as properties and methods at run time, inste
...
...stom dynamic object that searches the contents of a text file. Text to search for is specified by the name of a dynamic property. For example, if calling code specifies `dynamicFile.Sample`, the dynamic class returns a generic list of strings that contains all of the lines from the file that begin with "Sample". The search is case-insensitive. The dynamic class also supports two optional arguments. The first argument is a search option enum value that specifies that the dynamic class should search for matches at the start of the line, the end of the line, or anywhere in the line. The second argument specifies that the dynamic class should trim leading and trailing spaces from each line before searching. For example, if calling code specifies `dynamicFile.Sample(StringSearchOption.Contains)`, the dynamic class searches for "Sample" anywhere in a line. If calling code specifies `dynamicFile.Sample(StringSearchOption.StartsWith, false)`, the dynamic class searches for "Sample" at the start of each line, and does not remove leading and trailing spaces. The default behavior of the dynamic class is to search for a match at the start of each line and to remove leading and trailing spaces.
```
Comment (gewarren) [inline docs/csharp/programming-guide/types/walkthrough-creating-and-using-dynamic-objects.md:41]:
```suggestion
The first walkthrough defines a custom dynamic object that searches the contents of a text file. A dynamic property specifies the text to search for. For example, if calling code specifies `dynamicFile.Sample`, the dynamic class returns a generic list of strings that contains all of the lines from the file that begin with "Sample". The search is case-insensitive. The dynamic class also supports two optional arguments. The first argument is a search option enum value that specifies that the dynamic class should search for matches at the start of the line, the end of the line, or anywhere in the line. The second argument specifies that the dynamic class should trim leading and trailing spaces from each line before searching. For example, if calling code specifies `dynamicFile.Sample(StringSearchOption.Contains)`, the dynamic class searches for "Sample" anywhere in a line. If calling code specifies `dynamicFile.Sample(StringSearchOption.StartsWith, false)`, the dynamic class searches for "Sample" at the start of each line, and does not remove leading and trailing spaces. The default behavior of the dynamic class is to search for a match at the start of each line and to remove leading and trailing spaces.
```

Comment (gewarren) [reply docs/csharp/programming-guide/types/walkthrough-creating-and-using-dynamic-objects.md:41]: I just changed the second sentence.

File: docs/csharp/programming-guide/types/walkthrough-creating-and-using-dynamic-objects.md [hunk 3]
```diff
@@ -21,76 +20,88 @@ Dynamic objects expose members such as properties and methods at run time, inste
...
+1. In the **Configure your new project** dialog, enter `DynamicSample` for the **Project name**, and then select **Next**.
 
-3. In the **New Project** dialog box, in the **Project Types** pane, make sure that **Windows** is selected. Select **Console Application** in the **Templates** pane. In the **Name** box, type `DynamicSample`, and then click **OK**. The new project is created.
+1. In the **Additional information** dialog, select **.NET 5.0 (Current)** for the **Target Framework**, and then select **Create**.
 
-4. Right-click the DynamicSample project and point to **Add**, and then click **Class**. In the **Name** box, type `ReadOnlyFile`, and then click **OK**. A new file is added that contains the ReadOnlyFile class.
+   The new project is created.
 
-5. At the top of the ReadOnlyFile.cs or ReadOnlyFile.vb file, add the following code to import the <xref:System.IO?displayProperty=nameWithType> and <xref:System.Dynamic?displayProperty=nameWithType> namespaces.
+1. Right-click the DynamicSample project and select **Add** > **Class**. In the **Name** box, type `ReadOnlyFile`, and then select **Add**.
```
Comment (gewarren) [inline docs/csharp/programming-guide/types/walkthrough-creating-and-using-dynamic-objects.md:57]: In Solution Explorer?

File: docs/csharp/programming-guide/types/walkthrough-creating-and-using-dynamic-objects.md [hunk 4]
```diff
@@ -21,76 +20,88 @@ Dynamic objects expose members such as properties and methods at run time, inste
...
     [!code-vb[VbDynamicWalkthrough#7](~/samples/snippets/visualbasic/VS_Snippets_VBCSharp/vbdynamicwalkthrough/vb/readonlyfile.vb#7)]
 
-12. Save and close the file.
+1. Save and close the file.
 
-#### To create a sample text file
+### To create a sample text file
 
-1. Right-click the DynamicSample project and point to **Add**, and then click **New Item**. In the **Installed Templates** pane, select **General**, and then select the **Text File** template. Leave the default name of TextFile1.txt in the **Name** box, and then click **Add**. A new text file is added to the project.
+1. Right-click the DynamicSample project and select **Add** > **New Item**. In the **Installed Templates** pane, select **General**, and then select the **Text File** template. Leave the default name of *TextFile1.txt* in the **Name** box, and then click **Add**. A new text file is added to the project.
```
Comment (gewarren) [inline docs/csharp/programming-guide/types/walkthrough-creating-and-using-dynamic-objects.md:102]: In Solution Explorer?

**Review by gewarren — APPROVED** [review]
Looks good!

File: docs/csharp/programming-guide/types/walkthrough-creating-and-using-dynamic-objects.md [hunk 5]
```diff
@@ -107,51 +118,55 @@ The first project that you create in this walkthrough defines a custom dynamic o
...
+1. In **Solution Explorer**, double-click the *Program.vb* file if you're using Visual Basic or the *Program.cs* file if you're using Visual C#.
 
-2. Add the following code to the Main procedure to create an instance of the `ReadOnlyFile` class for the TextFile1.txt file. The code uses late binding to call dynamic members and retrieve lines of text that contain the string "Customer".
+2. Add the following code to the `Main` procedure to create an instance of the `ReadOnlyFile` class for the *TextFile1.txt* file. The code uses late binding to call dynamic members and retrieve lines of text that contain the string "Customer".
 
      [!code-csharp[VbDynamicWalkthrough#8](~/samples/snippets/csharp/VS_Snippets_VBCSharp/vbdynamicwalkthrough/cs/program.cs#8)]
-     [!code-vb[VbDynamicWalkthrough#8](~/samples/snippets/visualbasic/VS_Snippets_VBCSharp/vbdynamicwalkthrough/vb/module1.vb#8)]
+     [!code-vb[VbDynamicWalkthrough#8](~/samples/snippets/visualbasic/VS_Snippets_VBCSharp/vbdynamicwalkthrough/vb/Program.vb#8)]
 
 3. Save the file and press CTRL+F5 to build and run the application.
```
Comment (gewarren) [inline docs/csharp/programming-guide/types/walkthrough-creating-and-using-dynamic-objects.md:132]:
```suggestion
3. Save the file and press <kbd>Ctrl</kdb>+<kbd>F5</kbd> to build and run the application.
```

File: docs/csharp/programming-guide/types/walkthrough-creating-and-using-dynamic-objects.md [hunk 6]
```diff
@@ -107,51 +118,55 @@ The first project that you create in this walkthrough defines a custom dynamic o
...
 
-8. After the code to load the random.py module, add the following code to create an array of integers. The array is passed to the `shuffle` method of the random.py module, which randomly sorts the values in the array.
+1. After the code to load the random.py module, add the following code to create an array of integers. The array is passed to the `shuffle` method of the random.py module, which randomly sorts the values in the array.
 
      [!code-csharp[VbDynamicWalkthroughIronPython#3](~/samples/snippets/csharp/VS_Snippets_VBCSharp/vbdynamicwalkthroughironpython/cs/program.cs#3)]
-     [!code-vb[VbDynamicWalkthroughIronPython#3](~/samples/snippets/visualbasic/VS_Snippets_VBCSharp/vbdynamicwalkthroughironpython/vb/module1.vb#3)]
+     [!code-vb[VbDynamicWalkthroughIronPython#3](~/samples/snippets/visualbasic/VS_Snippets_VBCSharp/vbdynamicwalkthroughironpython/vb/Program.vb#3)]
 
-9. Save the file and press CTRL+F5 to build and run the application.
+1. Save the file and press CTRL+F5 to build and run the application.
```
Comment (gewarren) [inline docs/csharp/programming-guide/types/walkthrough-creating-and-using-dynamic-objects.md:169]:
```suggestion
1. Save the file and press <kbd>Ctrl</kdb>+<kbd>F5</kbd> to build and run the application.
```

**Comment by tdykstra** [comment]
Thanks for the review & suggestions @gewarren !

**Pull request merged**